### ISSUE TITLE
Criteria is not accessible, when :id is invalid

### ISSUE THREAD

**Issue by f1nnix** [dchester/epilogue#88]
Refrenced to #85 

My API provides aliases for id. /users/me/ === /users/5, if current user's ID is 5. So, I've tried such approach:

```js
users.read.fetch.before(function(req, res, context) {
  var userId = req.params.user_id
  
  if (userId === "me") {
    context.criteria.id = req.user.id
  }
  return context.continue;
})
```

Not working. Any ideas?

> 2) this should work, can you please open a new issue so we can discuss this if there is an actual problem here.

```js
var milestones = {
  users: require('./users.js')
}

var users = epilogue.resource({
  model: db.User,
  associations: true
});

users.use(milestones.users)
```
```js
// ./users.js
module.exports = {
  read: {
    start: function(req, res, context) {
      console.log(context);
      return context.continue;
    },
    auth: function(req, res, context) {
      console.log(context);
      return context.continue;
    },
    fetch: {
      before: function(req, res, context) {
        console.log(context);
        return context.continue;
      }
    }
  }
}
```
## Tesing

GET /api/v0/users/me

### Output

```
{ instance: undefined,
  criteria: {},
  attributes: {},
  skip: [Function],
  stop: [Function],
  continue: [Function],
  error: [Function] }
{ instance: undefined,
  criteria: {},
  attributes: {},
  skip: [Function],
  stop: [Function],
  continue: [Function],
  error: [Function] }
{ instance: undefined,
  criteria: {},
  attributes: {},
  skip: [Function],
  stop: [Function],
  continue: [Function],
  error: [Function] }
```

### Expected

```json
criteria: {
  "id": "me"
}
```
  
**UPD**: Also tried to access to req object directly:

```js
auth: authMiddleware.E.ensureAuthenticated,
fetch: {
  before: function(req, res, context) {
    console.log(context);
    // :me
    if (req.params.id === 'me') {
      context.criteria.id = req.user.id;
    }
    console.log(context);
    return context.continue;
  }
}
```
  
Response is:

```json
{
  "message": "\"me\" is not a valid integer",
  "errors": []
}
```

The request doen't respect `criteria: {}` object, does it?

**Comment by mbroadst** [comment]
this looks like its an actual bug [here](https://github.com/dchester/epilogue/blob/master/lib/Controllers/read.js#L25). The logic needs to be something like: `if (attribute in req.params && !!criteria[attribute])`

**Comment by mbroadst** [comment]
fixed in 470abb2e04f2b08bdd0bbaeb124c9a7ce3843c31
